package/base-files/files/bin/ipcalc.sh

```diff
@@ -44,28 +43,36 @@ BEGIN {
 	}
 
 	network=and(ipaddr,netmask)
+	prefix=32-bitcount(compl32(netmask))
 	broadcast=or(network,compl32(netmask))
 
-	start=or(network,and(ip2int(ARGV[3]),compl32(netmask)))
-	limit=network+1
-	if (start<limit) start=limit
-
-	end=start+ARGV[4]
-	limit=or(network,compl32(netmask))-1
-	if (end>limit) end=limit
-
 	print "IP="int2ip(ipaddr)
 	print "NETMASK="int2ip(netmask)
 	print "BROADCAST="int2ip(broadcast)
 	print "NETWORK="int2ip(network)
-	print "PREFIX="32-bitcount(compl32(netmask))
+	print "PREFIX="prefix
 
 	# range calculations:
 	# ipcalc <ip> <netmask> <start> <num>
 
-	if (ARGC > 3) {
-		print "START="int2ip(start)
-		print "END="int2ip(end)
+	if (ARGC <= 3)
+		exit(0)
```

Comment:
https://github.com/openwrt/openwrt/blob/98bb26f9f762408e42bd8a906f0eb01c41ada10a/package/network/ipv6/6rd/files/6rd.sh#L44
https://github.com/openwrt/packages/blob/6e0f77acbeef4134d750bc8cfd11d187359ea348/net/isc-dhcp/files/dhcpd.init#L409

Reply:
Hi :-)

Could you say a bit more about what makes this line interesting?

netifd and 6rd both use ipcalc.sh with a single argument - that still works:

```
$ ./ipcalc "192.168.0.24/24"
IP=192.168.0.24
NETMASK=255.255.255.0
BROADCAST=192.168.0.255
NETWORK=192.168.0.0
PREFIX=24
```

Reply:
I'm sorry. I thought it won't work for other packages.

Reply:
No problem! Rather have too many checks than too few ;-)